### PR TITLE
BRO Mishra, Claimed by Gix + Support

### PR DESCRIPTION
**Pull request by Simisays**
Adding the param to make the melded card enter tapped was pretty straightforward but I'm having difficulty adding the "attacking" param. Help would be appreciated

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/mishra_claimed_by_gix_mishra_lost_to_phyrexia.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
+Name:Mishra, Claimed by Gix
+ManaCost:2 B R
+Types:Legendary Creature Phyrexian Human Artificer
+PT:3/5
+T:Mode$ AttackersDeclaredOneTarget | AttackedTarget$ Player,Planeswalker | ValidAttackers$ Creature.YouCtrl |  Execute$ TrigDrain | TriggerZones$ Battlefield | TriggerDescription$ Whenever you attack, each opponent loses X life and you gain X life, where X is the number of attacking creatures. If CARDNAME and a creature named Phyrexian Dragon Engine are attacking, and you both own and control them, exile them, then meld them into Mishra, Lost to Phyrexia. It enters the battlefield tapped and attacking
+SVar:TrigDrain:DB$ LoseLife | Defined$ Player.Opponent | LifeAmount$ X | SubAbility$ DBGainLife
+SVar:DBGainLife:DB$ GainLife | LifeAmount$ X | SubAbility$ Meld 
+SVar:Meld:DB$ Meld | IsPresent$ Card.Self+YouOwn+Attacking | IsPresent2$ Creature.YouCtrl+YouOwn+namedPhyrexian Dragon Engine+Attacking | Name$ Mishra, Lost to Phyrexia | EntersTapped$ True | Attacking$ True | Primary$ Mishra, Claimed by Gix | Secondary$ Phyrexian Dragon Engine
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/mishra_claimed_by_gix_mishra_lost_to_phyrexia.txt:8]: capital is wrong, `attacking`

File: forge-game/src/main/java/forge/game/ability/effects/MeldEffect.java [hunk 2]
```diff
@@ -47,12 +47,19 @@ public void resolve(SpellAbility sa) {
...
 
         for (Card c : exiled) {
+        boolean attacking = sa.hasParam("Attacking");
             if (c.isToken() || c.getCloneOrigin() != null) {
                 // Neither of these things
                 return;
             } else if (!c.isInZone(ZoneType.Exile)) {
                 return;
             }
+            if (sa.hasParam("EntersTapped")) {
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/ability/effects/MeldEffect.java:57]: Name it "Tapped" for consistency

File: forge-game/src/main/java/forge/game/ability/effects/MeldEffect.java [hunk 3]
```diff
@@ -47,12 +47,19 @@ public void resolve(SpellAbility sa) {
...
+                c.setTapped(true);
+            }
+            if (sa.hasParam("Attacking")) {
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/ability/effects/MeldEffect.java:60]: Do this at the **bottom**:
`        if (addToCombat(movedCard, movedCard.getController(), sa, "Attacking", "Blocking")) {
            game.updateCombatForView();
            game.fireEvent(new GameEventCombatChanged());
        }`

movedCard is what `changeZone()` returns

File: forge-gui/res/cardsfolder/upcoming/mishra_claimed_by_gix_mishra_lost_to_phyrexia.txt [hunk 4]
```diff
@@ -0,0 +1,32 @@
+Name:Mishra, Claimed by Gix
+ManaCost:2 B R
+Types:Legendary Creature Phyrexian Human Artificer
+PT:3/5
+T:Mode$ AttackersDeclaredOneTarget | AttackedTarget$ Player,Planeswalker | ValidAttackers$ Creature.YouCtrl |  Execute$ TrigDrain | TriggerZones$ Battlefield | TriggerDescription$ Whenever you attack, each opponent loses X life and you gain X life, where X is the number of attacking creatures. If CARDNAME and a creature named Phyrexian Dragon Engine are attacking, and you both own and control them, exile them, then meld them into Mishra, Lost to Phyrexia. It enters the battlefield tapped and attacking
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/mishra_claimed_by_gix_mishra_lost_to_phyrexia.txt:5]: wrong trigger, not OneTarget

**Comment by Simisays** [comment]
@tool4ever  Thanks for the help, I think I got it working !
image

File: forge-game/src/main/java/forge/game/ability/effects/MeldEffect.java [hunk 5]
```diff
@@ -54,12 +55,21 @@ public void resolve(SpellAbility sa) {
...
+        }
 
         primary.changeToState(CardStateName.Meld);
         primary.setBackSide(true);
         primary.setMeldedWith(secondary);
         PlayerZoneBattlefield bf = (PlayerZoneBattlefield)controller.getZone(ZoneType.Battlefield);
         game.getAction().changeZone(primary.getZone(), bf, primary, 0, sa);
         bf.addToMelded(secondary);
+        
+        if (addToCombat(primary, primary.getController(), sa, "Attacking", "Blocking")) {
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/ability/effects/MeldEffect.java:70]: Like I said you should call this on the movedCard from changeZone

File: forge-game/src/main/java/forge/game/ability/effects/MeldEffect.java [hunk 6]
```diff
@@ -66,10 +65,11 @@ public void resolve(SpellAbility sa) {
         PlayerZoneBattlefield bf = (PlayerZoneBattlefield)controller.getZone(ZoneType.Battlefield);
         game.getAction().changeZone(primary.getZone(), bf, primary, 0, sa);
```
[the remark this answers is not in the document]
Comment (tool4ever) [reply forge-game/src/main/java/forge/game/ability/effects/MeldEffect.java:67]: You can't just call changeZone twice, remove the first one

**Review by tool4ever — APPROVED** [review]

**Pull request merged**